From 15983e91ce47af05764cd0ae0c44d982d9039eda Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Tue, 16 Oct 2018 16:43:02 +0800
Subject: [PATCH 1/4] driver: ata: libata-core: init "rcw_base" &
 "serdes_base" to avoid warning

arch/arm64/include/asm/io.h:175:20: warning: 'rcw_base' may be used uninitialized in this function [-Wmaybe-uninitialized]
 #define iounmap __iounmap
                    ^~~~~~~~~
drivers/ata/libata-core.c:4025:16: note: 'rcw_base' was declared here
  void __iomem *rcw_base;
                ^~~~~~~~

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/ata/libata-core.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/ata/libata-core.c b/drivers/ata/libata-core.c
index 5f92f7f..b255194 100644
--- a/drivers/ata/libata-core.c
+++ b/drivers/ata/libata-core.c
@@ -4021,8 +4021,8 @@ int sata_link_hardreset(struct ata_link *link, const unsigned long *timing,
 	struct ahci_qoriq_priv *qoriq_priv = hpriv->plat_data;
 	bool lx2160a_workaround = (qoriq_priv->type == AHCI_LX2160);
 
-	void __iomem *serdes_base;
-	void __iomem *rcw_base;
+	void __iomem *serdes_base = NULL;
+	void __iomem *rcw_base = NULL;
 	u32 scontrol;
 	int rc;
 	int val;
-- 
1.7.9.5

